File: finance-api/src/main/java/com/financesystem/finance_api/modules/platform/servicepayments/domain/model/ServiceProvider.java
package com.financesystem.finance_api.modules.platform.servicepayments.domain.model;

import java.time.Instant;
import java.util.UUID;

public record ServiceProvider(
        UUID id,
        String code,
        String name,
        ServiceProviderCategory category,
        String serviceCustomerCodeLabel,
        ServiceProviderStatus status,
        Instant createdAt,
        Instant updatedAt
) {
}

File: finance_mobile/lib/domain/entities/service_provider.dart
class ServiceProvider {
  final String id;
  final String code;
  final String name;
  final String category;
  final String? serviceCustomerCodeLabel;
  final String? status;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const ServiceProvider({
    required this.id,
    required this.code,
    required this.name,
    required this.category,
    this.serviceCustomerCodeLabel,
    this.status,
    this.createdAt,
    this.updatedAt,
  });

  bool get isActive => (status ?? '').toUpperCase() == 'ACTIVE';

  String get displayCategory {
    return category.replaceAll('_', ' ');
  }
}
